src/core/config/config.go

```diff
@@ -149,6 +152,17 @@ func RegisterFlags() {
 		if err := Viper.BindPFlag(strings.ReplaceAll(flag.Name, "-", "_"), fs.Lookup(flag.Name)); err != nil {
 			return
 		}
+
+		oldKey := strings.ToUpper(OldEnvPrefix + "_" + strings.ReplaceAll(flag.Name, "-", "_"))
+		newKey := strings.ToUpper(NewEnvPrefix + "_" + strings.ReplaceAll(flag.Name, "-", "_"))
+
+		if os.Getenv(oldKey) != "" && os.Getenv(newKey) == "" {
```

Comment:
What about the (admittedly unlikely) case where both `oldKey` **and** `newKey` are set?

Reply:
If both oldKey and newKey are set we assume the newKey is the one with the correct value and ignore the value from the old key